pkg/apis/core/v1beta1/defaults/defaults_test.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package defaults
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	apiextv1b1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"sigs.k8s.io/kubefed/pkg/controller/util"
+)
+
+func TestSetDefaultKubeFedConfig(t *testing.T) {
+	type KubeFedConfigComparison struct {
+		original *v1beta1.KubeFedConfig
+		modified *v1beta1.KubeFedConfig
+	}
+
+	successCases := map[string]KubeFedConfigComparison{}
+
+	// ControllerDuration
+	availableDelayKFC := defaultKubeFedConfig()
```

Comment:
(No action required) I find these kinds of boilerplate-heavy tests really hard to review. I guess there simply aren't enough of these tests to justify a more data-driven approach, but I would prefer to avoid creating any more of these than absolutely necessary.